### DISCUSSION TITLE
How can I add react component in the block manager?

### DISCUSSION THREAD

**Discussion by ShohanMir** [GrapesJS/grapesjs#5250] [Q&A]
Is there any updated example to add react component in the block manager? I've tried a way but it's not working.

**Comment by eschiebel** [comment]
^that example solution doesn't work in general because `base-react-component`'s code is executed in the application's window, then the resulting element is appended to `<div data-gjs-type="wrpper">` within the grapesjs editor's iframe where the component's code does not exist. Is there a way to defer rendering the react component until it's been added to the editor's iframe?
